From 9e736adc21e2e347bfa0cf2e3edc5d28fb1b4832 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <martin.abente.lahaye@gmail.com>
Date: Fri, 24 Jan 2020 11:51:05 -0300
Subject: [PATCH 4/5] fix-greeting

---
 activity.py | 19 ++++++-------------
 1 file changed, 6 insertions(+), 13 deletions(-)

diff --git a/activity.py b/activity.py
index 260f5bb..335f25b 100644
--- a/activity.py
+++ b/activity.py
@@ -372,19 +372,12 @@ class SpeakActivity(SugarCompatibleActivity):
 
         presenceService = presenceservice.get_instance()
         self.owner = presenceService.get_owner()
-        if self._first_time:
-            # say hello to the user
-            if self._tablet_mode:
-                self._entry.props.text = _('Hello %s.') \
-                    % self.owner.props.nick
-            self.face.say_notification(_('Hello %s. Please Type something.')
-                                       % self.owner.props.nick)
-        else:
-            if self._tablet_mode:
-                self._entry.props.text = _('Welcome back %s.') \
-                    % self.owner.props.nick
-            self.face.say_notification(_('Welcome back %s.')
-                                       % self.owner.props.nick)
+        # say hello to the user
+        if self._tablet_mode:
+            self._entry.props.text = _('Hello %s.') \
+                % self.owner.props.nick
+        self.face.say_notification(_('Hello %s. Please Type something.')
+                                   % self.owner.props.nick)
         self._set_idle_phrase(speak=False)
         self._first_time = False
 
-- 
2.23.0

