diff --git a/activity.py b/activity.py
index 5dacc14..260f5bb 100644
--- a/activity.py
+++ b/activity.py
@@ -84,6 +84,9 @@ from faceselect import FaceSelector
 
 import speech
 
+from sugarapp.widgets import SugarCompatibleActivity
+from sugarapp.widgets import DesktopOpenChooser
+
 SERVICE = 'org.sugarlabs.Speak'
 IFACE = SERVICE
 PATH = '/org/sugarlabs/Speak'
@@ -166,7 +169,7 @@ def _is_tablet_mode():
     return False
 
 
-class SpeakActivity(activity.Activity):
+class SpeakActivity(SugarCompatibleActivity):
     def __init__(self, handle):
         super(SpeakActivity, self).__init__(handle)
 
@@ -328,6 +331,8 @@ class SpeakActivity(activity.Activity):
         else:
             # we are creating the activity
             self.connect('shared', self._shared_cb)
+        # hide chat mode since it won't work
+        self._mode_chat.hide()
 
     def _toolbar_expanded(self):
         if self._activity_button.is_expanded():
@@ -748,23 +753,19 @@ class SpeakActivity(activity.Activity):
         return facebar
 
     def _photo_face_cb(self, widget):
-        chooser = ObjectChooser(parent=self,
-                                what_filter=mime.GENERIC_TYPE_IMAGE)
-
-        result = chooser.run()
-        if result == Gtk.ResponseType.ACCEPT:
-            jobject = chooser.get_selected_object()
-            if jobject and jobject.file_path:
-                selector = FaceSelector(jobject.file_path)
-                selector.connect('face-processed',
-                                 self._photo_face_processed_cb)
-                selector.connect('cancel', self._photo_face_cancel_cb)
-                self._notebook.append_page(selector, Gtk.Label(''))
-                selector.show()
-
-                num = self._notebook.page_num(selector)
-                self._notebook.set_current_page(num)
-        chooser.destroy()
+        chooser = DesktopOpenChooser(self.get_toplevel())
+        chooser.add_filter('.png', 'png')
+
+        filename = chooser.get_filename()
+        selector = FaceSelector(filename)
+        selector.connect('face-processed',
+                         self._photo_face_processed_cb)
+        selector.connect('cancel', self._photo_face_cancel_cb)
+        self._notebook.append_page(selector, Gtk.Label(''))
+        selector.show()
+
+        num = self._notebook.page_num(selector)
+        self._notebook.set_current_page(num)
 
     def _photo_face_processed_cb(self, widget, *face_data):
         lighter = style.Color(self._colors[_lighter_color(self._colors)])
diff --git a/brain.py b/brain.py
index bbae17f..9ad6ba5 100644
--- a/brain.py
+++ b/brain.py
@@ -64,8 +64,7 @@ _kernel_voice = None
 
 
 def _get_age():
-    settings = Gio.Settings('org.sugarlabs.user')
-    birth_timestamp = settings.get_int('birth-timestamp')
+    birth_timestamp = None
     if birth_timestamp is None or birth_timestamp == 0:
         return 8
     else:

